apps/documents/serializers.py

```diff
@@ -68,6 +67,10 @@ class Meta:
             exclude = ('shipment',)
         meta_fields = ('presigned_s3',)
 
+    def validate(self, attrs):
+        attrs['shipment_id'] = self.context['shipment_id']
```

Comment:
I think this logic should exist in create() not validate(). Just pass self.context['shipment_id'] directly in to the Document.objects.create() call.